app/ui/src/app/customizations/api-connector/api-connector.service.ts

```diff
@@ -1,23 +1,43 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
 
-import { ApiConnectorData, ApiConnectors } from './api-connector.models';
 import { ApiHttpService } from '@syndesis/ui/platform';
+import {
+  ApiConnectors,
+  ApiConnectorData, CustomConnectorRequest
+} from './api-connector.models';
 
 @Injectable()
 export class ApiConnectorService {
   constructor(private apiHttpService: ApiHttpService) { }
 
-  getApiConnector(id): Observable<ApiConnectorData> {
+  getApiConnector(id: string): Observable<ApiConnectorData> {
     return this.apiHttpService
       .setEndpointUrl('getApiConnectorDetails', { id })
       .get<ApiConnectorData>();
   }
 
-  list(): Observable<ApiConnectors> {
+  getApiConnectorList(): Observable<ApiConnectors> {
     return this.apiHttpService
       .setEndpointUrl('getApiConnectorList', { template: 'swagger-connector-template'})
-      .get<{ items: ApiConnectors }>() // TODO: Provide a better typing T here
+      .get<{ items: ApiConnectors }>() // TODO: Provide a better <T> typing here
```

Comment:
TODO: No more TODOS! 😭

Reply:
TODO: No more technical debt. I agree that this is not ideal, but tracking stuff that needs to be catered (and cannot be fixed at this stage) is a good way to embrace the [The boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule) and prevent the introduction of more technical debt in the future.

Also, doing this is a good way to prevent other programmers like: "Hey, this needs to be fixed. Can you do it now?"